docs/antora/modules/ROOT/pages/Extending_Mill.adoc

```diff
@@ -149,14 +150,72 @@ def generatedHtml = T {
   ).render  
 }
 ----
+<1> Import the scalatags library from Mavel Central repository.
+<2> Creates the `generatedHtml` target which is used here to generate a simple Hello World HTML document. It can be used however you would like: written to a file, further processed, etc.
 
-This creates the `generatedHtml` target which can then be used however you would like: written to a file, further processed, etc.
-
-If you want to publish re-usable libraries that _other_ people can use in their builds, simply publish your code as a library to maven central.
+Please also read the next section xref:_using_mill_plugins_import_ivy[]].
 
 For more information, see Ammonite's
 http://ammonite.io/#import$ivy[Ivy Dependencies documentation].
 
+
+== Using Mill Plugins (`import $ivy`)
+
+Mill plugins are ordinary jars and are loaded as any other external dependency with xref:_import_ivy[`import $ivy`].
+
+Mill plugins are typically bound to a specific version or version range of Mill.
+To ease the use of the correct versions and avoid runtime issues (caused by binary incompatible plugins, which are hard to debug) you can apply one of the following techniques:
+
+=== Use the specific Mill Binary Platform notation
+
+[source,scala]
+----
+// for classic Scala dependencies
+import $ivy.`<group>::<plugin>::<version>` // <1>
+// for dependencies specific to the exact Scala version
+import $ivy.`<group>:::<plugin>::<version>` // <2>
+----
+<1> This is equivalent to
++
+[source,scala]
+----
+import $ivy.`<group>::<plugin>_mill$MILL_BIN_PLATFORM:<version>`
+----
+<1> This is equivalent to
++
+[source,scala]
+----
+import $ivy.`<group>:::<plugin>_mill$MILL_BIN_PLATFORM:<version>`
+----
+
+
+=== Use special placeholders in your `import $ivy`
+
+`$MILL_VERSION` ::
++
+--
+to substitute the currently used Mill version.
+This is typical required for Mill contrib modules, which are developed in the Mill repository and highly bound to the current Mill version.
+
+.Example: Use `mill-contrib-bloop` plugin matching the current Mill version
+----
+import $ivy:`com.lihaoyi:mill-contrib-bloop:$MILL_VERSION`
```

Comment:
Shouldn't this be:
```suggestion
import $ivy.`com.lihaoyi:mill-contrib-bloop:$MILL_VERSION`
```
?

Reply:
You're absolutely right. I plan to also pack-port PR #1526 and therefore will keep this as-is to avoid clashes.